terraform/main.tf

```diff
@@ -38,6 +38,10 @@ provider "sentry" {
   # Must set SENTRY_AUTH_TOKEN envvar
 }
 
+provider "improvmx" {
+  # Must set IMPROVMX_API_TOKEN envvar
+}
+
```

Comment:
This is a question for my own education, but why list all the provider blocks in one place, rather than in the file where it's actually used? Maybe because some providers get used in multiple files, in which case it makes sense to "refactor" the provider block to a known place like main.tf?

(I have a similar question about the required_providers block in versions.tf, but I suspect the answer would be the same.)

Reply:
I think it's just the convention for Terraform. I've never seen it done any other way. I also like it because you can quickly see what providers a module is using.